### PR TITLE
HPCC-26627 Add CookieConsent form component

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage - is this still draft?

**Comment by jbrundage** [comment]
@GordonSmith it's ready for review

File: esp/src/src-react/components/Frame.tsx [hunk 1]
```diff
@@ -40,6 +41,10 @@ export const DevFrame: React.FunctionComponent<DevFrameProps> = () => {
             ;
     }, [isDark]);
 
+    const consent = !!(getCookie("user_cookie_consent"));
```
Comment (GordonSmith) [inline esp/src/src-react/components/Frame.tsx:44]: Should use `LocalStorage` from `src/KeyValStore.ts`

**Review by GordonSmith — CHANGES_REQUESTED** [review]
Can you also review:  https://www.cookiepro.com/knowledge/cookie-banner-guidelines/ specifically the GDPR one?

**Review by GordonSmith — CHANGES_REQUESTED** [review]
I am not seeing the GDPR verbage?

File: esp/src/src-react/components/Frame.tsx [hunk 2]
```diff
@@ -40,6 +43,16 @@ export const DevFrame: React.FunctionComponent<DevFrameProps> = () => {
             ;
     }, [isDark]);
 
+    sessionKeyValStore().get("user_cookie_consent")
```
Comment (GordonSmith) [inline esp/src/src-react/components/Frame.tsx:46]: This should be a "userKeyValStore" otherwise it nag the user every session.

Comment (GordonSmith) [reply esp/src/src-react/components/Frame.tsx:46]: Also there isa React hook for this:  "useUserStore"

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — CHANGES_REQUESTED** [review]
Switch from sessionKeyValStore to useUserStore.

**Comment by richardkchapman** [comment]
Should this target 8.6.x

**Comment by GordonSmith** [comment]
@richardkchapman it should now.

**Comment by richardkchapman** [comment]
@jbrundage Please rebase and retarget to 8.6

File: esp/src/src-react/components/Frame.tsx [hunk 3]
```diff
@@ -40,6 +43,16 @@ export const DevFrame: React.FunctionComponent<DevFrameProps> = () => {
             ;
     }, [isDark]);
 
+    userKeyValStore().get("user_cookie_consent")
+        .then((resp)=>{
+            setShowCookieConsent(resp === "1");
+        })
+        ;
```
Comment (GordonSmith) [inline esp/src/src-react/components/Frame.tsx:50]: This should be inside a `React.useEffect` (as it has side effects)

File: esp/src/src-react/components/Frame.tsx [hunk 4]
```diff
@@ -40,6 +43,16 @@ export const DevFrame: React.FunctionComponent<DevFrameProps> = () => {
...
+
+    const consentComponent = <CookieConsent showCookieConsent={showCookieConsent} onApply={(n: boolean) => {
+        userKeyValStore().set("user_cookie_consent", n ? "1" : "0");
+    }}/>;
```
Comment (GordonSmith) [inline esp/src/src-react/components/Frame.tsx:54]: You can move this into the full return structure

File: esp/src/src-react/components/forms/CookieConsent.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    setShowCookieConsent?: (_: boolean) => void;
+}
+
+export const CookieConsent: React.FunctionComponent<CookieConsentProps> = ({
+    onApply = () => {},
+    showCookieConsent,
+    setShowCookieConsent = () => {}
+}) => {
+
+    const closeCookieConsent = () => setShowCookieConsent(false);
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/CookieConsent.tsx:18]: This can be moved inline into line 20

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/Frame.tsx [hunk 6]
```diff
@@ -31,6 +34,12 @@ export const DevFrame: React.FunctionComponent<DevFrameProps> = () => {
 
         router.resolve(hashHistory.location).then(setBody);
 
+        userKeyValStore().get("user_cookie_consent")
+            .then((resp)=>{
+                setShowCookieConsent(resp === "1");
+            })
+            ;
+
```
Comment (GordonSmith) [inline esp/src/src-react/components/Frame.tsx:42]: Nees to be inside a `React.useEffect`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**